src/appverse-translate/module.js

```diff
@@ -57,57 +57,39 @@
      * after the translation
      *
      * @param  {array}      $delegate       The original instance (provided by decorator)
-     * @param  {function}   translateFilter
+     * @param  {Object}     I18N_CONFIG     The I18N_CONFIG appverse object
      * @return {array}                      The modified delegate object
      */
-    function decorateTranslateDirective($delegate, translateFilter) {
+    function decorateTranslateDirective($delegate, I18N_CONFIG) {
 
-        // Get the original directive and its linking function
+        // Get the original directive and its compile function
         var directive = $delegate[0];
-        var originalLinkFunction = directive.link;
-
-        var newLinkFunction = function (scope, $element, attr, ctrl) {
-
-            // Get the element's html and replaces the text to be translated
-            // by a placeholder '%%text%%', so that we can later replace this
-            // with the translated string
-            var text = $element.text();
-            var html = $element.html();
-            var htmlOnlyTags = html.replace(text, '%%text%%');
-
-            // First we call the original linking function
-            // and afterwards we override the '$on' and '$watch' events
-            // to maintain html tags.
-            originalLinkFunction.apply(this, [scope, $element, attr, ctrl]);
-
-            scope.$on('$translateChangeSuccess', function () {
-                translateElement();
-            });
-
-            scope.$watch('[translationId, interpolateParams]', function () {
-                if (scope.translationId) {
-                    translateElement();
+        var originalCompile = directive.compile;
+
+        directive.compile = function compile(tElement, tAttr, transclude) {
+            var originalLink = originalCompile(tElement, tAttr);
+            return function newLink(scope, $element, attr, ctrl) {
+
+                if ($element.children().length) {
+                    // Get the element's html and replaces the text to be translated
+                    // by a placeholder '%%text%%', so that we can later replace this
+                    // with the translated string
+                    var text = $element.text();
+                    var html = $element.html();
+                    var htmlOnlyTags = html.replace(text, '%%text%%');
+
+                    scope.$watch(function () {
+                        var translatedText = $element[I18N_CONFIG.acceptHtml ? 'html' : 'text']();
```

Comment:
`I18N_CONFIG.acceptHtml` looks like a new option. It should be added to I18N_CONFIG constant.

Reply:
Actually this option has not been added on this pull-request.
https://github.com/Appverse/appverse-web-html5-core/blob/3efb6b97340c245f418a8d11b831dee1852edd7e/src/appverse-translate/module.js#L95